packages/sdk/src/getTransferData/getSourceData.ts

```diff
@@ -289,15 +336,23 @@ export async function getFee({
   throw new Error('Either contract or extrinsic must be provided');
 }
 
-export async function getContractFee(
-  balance: bigint,
-  config: ContractConfig,
-  decimals: number,
-  evmSigner: EvmSigner,
-): Promise<bigint> {
+export async function getContractFee({
+  balance,
+  config,
+  decimals,
+  evmSigner,
+  chain,
+}: {
+  balance: bigint;
+  config: ContractConfig;
+  decimals: number;
+  evmSigner: EvmSigner;
+  chain: EvmParachain;
+}): Promise<bigint> {
   const contract = createContract(
     config,
     evmSigner,
+    chain,
```

Comment:
if we're always passing `chain` it would not make sense to change the function definition to mark `chain` as not optional

Reply:
I think it's not optional: https://github.com/moonbeam-foundation/xcm-sdk/blob/riha/add-peaq/packages/sdk/src/getTransferData/getTransferData.utils.ts#L18

Reply:
isn't it this https://github.com/moonbeam-foundation/xcm-sdk/blob/riha/add-peaq/packages/sdk/src/contract/contract.factory.ts#L47 ?